### PR TITLE
HsYAML 0.2

### PR DESCRIPTION
**Pull request by vijayphoenix**
* Refactored code and modified dependence
* This works with the current [head](https://github.com/haskell-hvr/HsYAML/commit/d477f719d91d875042f5f4e0f118df51efe8f7f5) of the HsYAML repository.

### CODE REVIEW

**Comment by vijayphoenix** [comment]
* Functions for Encoding JSON Values
* New file `cabal.project`
* Updated dependencies and package's version to `0.2.0.0`
* Documentation

File: cabal.project [hunk 1]
```diff
@@ -0,0 +1,8 @@
+packages: .
+
+source-repository-package
```
Comment (vijayphoenix) [inline cabal.project]: Temporary source-repository-package.
I will remove it after HsYAML-0.2 is released.

File: src/Data/YAML/Aeson.hs [hunk 2]
```diff
@@ -141,3 +160,58 @@ scalarToValue (Y.SFloat x)   = Just $! J.Number (realToFrac x)
...
+{-# INLINE bsToStrict #-}
+bsToStrict :: BS.L.ByteString -> BS.ByteString
+#if MIN_VERSION_bytestring(0,10,0)
+bsToStrict = BS.L.toStrict
+#else
+bsToStrict = BS.concat . BS.L.toChunks
+#endif
+
+-- | @since 0.2.0
+instance ToYAML J.Value where
```
Comment (vijayphoenix) [inline src/Data/YAML/Aeson.hs]: Orphan instance: instance ToYAML Value
Shall I move this to HsYAML?

Comment (hvr) [reply src/Data/YAML/Aeson.hs]: No; that would require to have `HsYAML` depend on `aeson` which would then make having a separate HsYAML-aeson package pointless... :-)

simply add `{-# OPTIONS_GHC -fno-warn-orphans #-}` to the top of the file

File: HsYAML-aeson.cabal [hunk 3]
```diff
@@ -30,12 +30,16 @@ source-repository head
 library
   exposed-modules: Data.YAML.Aeson
   build-depends:
-    , HsYAML      ^>= 0.1.1.0
+    , HsYAML      ^>= 0.2
     , aeson       ^>= 1.4.0.0
     , base         >= 4.5 && < 4.13
     , bytestring  ^>= 0.9.2.1 || ^>= 0.10.0.2
+    , containers   >=0.4.2 && <0.7
+    , ordered-containers
```
Comment (hvr) [inline HsYAML-aeson.cabal]: missing version constraints for `ordered-containers`; but note that HsYAML-aeson currently supports base-4.5; and ordered-containers only supports base-4.5 in version 0.1.* (see https://matrix.hackage.haskell.org/#/package/ordered-containers)

what are we using `ordered-containers` for?

File: HsYAML-aeson.cabal [hunk 4]
```diff
@@ -30,12 +30,16 @@ source-repository head
...
     , mtl         ^>= 2.2.1
-    , text        ^>= 1.2.3.1
+    , scientific  ^>= 0.3.6.2
+    , text        ^>= 1.2.3
+    , unordered-containers >= 0.2 && < 0.3
```
Comment (hvr) [inline HsYAML-aeson.cabal]: Use `^>=`  operator

File: HsYAML-aeson.cabal (hunk 3 above)
Comment (vijayphoenix) [reply HsYAML-aeson.cabal]: forgot to remove it

**Pull request merged**